### DISCUSSION TITLE
Spec file elsewhere than project root

### DISCUSSION THREAD

**Discussion by kernc** [pyinstaller/pyinstaller#7087] [Polls]
First of all, PyInstaller was a breeze to figure out and seems to work beautifully on all relevant platforms — Thank you!

But is it just me or is there really no way to have the spec file outside project root (but instead, e.g., in ./installers/ dir that keeps other related resources including different spec files, version.rc etc.) and still be able to address all paths in the spec (such as `icon=`) relative to project root?

Something like:
```sh
$ cd my_project
# Since pyinstaller respects location of the spec file rather than CWD ...
$ pyinstaller installers/windows.spec --spec-cwd=.
```
?

**Comment by bwoodsend** [comment]
It's just you.

**Reply by kernc** [reply]
You mean, like, `cp installers/windows.spec .`? 😅

**Reply by bwoodsend** [reply]
The following generates a spec file which runs from an `installers` directory.
```fish
echo 'print("hello")' > test.py
mkdir installers
pyi-makespec --specpath installers/ test.py
pyinstaller installers/test.spec
```
As you've figured out, filenames are relative to the spec's location so all that differs is the filenames gain a leading `../`.
```diff
  
  
  a = Analysis(
-     ['test.py'],
      pathex=[],
      binaries=[],
      datas=[],
--- 5,11 ----
  
  
  a = Analysis(
+     ['../test.py'],
      pathex=[],
      binaries=[],
      datas=[],
```
Is that what you're asking? Reading your question again I think you may instead be asking if there's any way to have it ignore the spec's location in favour of the project root so that you don't have to modify all your filenames in the spec file?

**Reply by kernc** [reply]
Right, I see build and dist dirs remain placed relative to CWD, not specfile. Good, I followed your proposal and looks like it's all working. Slightly irritating that some resources are now prefixed with `"../"` wherease some are not (e.g. the version.rc.in template that I preprocess inside the _spec script itself_), but I can live with it. 😄 Many thanks for this simple and acceptable workaround!
